app/controllers/comments_controller.rb

```diff
@@ -28,22 +30,27 @@ def create
   def update
     respond_to do |format|
       if @comment.update(comment_params)
+        reference = @comment.root.commentable
         format.html { redirect_to :back, notice: 'Comment was successfully updated.' }
         format.json { render :show, status: :ok, location: @comment }
+        format.js { render 'reference_comments.js.erb', locals: {reference: reference} }
       else
         format.html { redirect_to :back, notice: 'Comment was not updated.' }
         format.json { render json: @comment.errors, status: :unprocessable_entity }
+        format.js { render 'reference_comments.js.erb', locals: {reference: reference} }
       end
     end
   end
 
   # DELETE /comments/1
   # DELETE /comments/1.json
   def destroy
+    reference = @comment.root.commentable
```

Comment:
I forgot to note earlier that this will stop working if we add another model as commentable